spec.html

```diff
@@ -32711,6 +32674,18 @@ <h1>Array.prototype.sort ( _comparefn_ )</h1>
           <p>The `sort` function is intentionally generic; it does not require that its *this* value be an Array object. Therefore, it can be transferred to other kinds of objects for use as a method.</p>
         </emu-note>
 
+        <emu-clause id="sec-objectissparse" aoid="ObjectIsSparse">
+          <h1>Runtime Semantics: ObjectIsSparse ( _obj_, _len_ )</h1>
```

Comment:
this seems more like it should be "ArraylikeIsSparse"?

Reply:
Maybe.

The spec doesn't actually define "array-like", and I don't think I want to contribute to the suggestion that it does. It'd be different if we were to define it...

Based on the evidence (in `CreateListFromArrayLike` and the preamble of `ToLength`), it seems like the only requirement on an array-like object is that it have a "length" property whose value is (or can be converted by various means to) an integer in the range [0, 2^53 - 1]. Or more specifically, that ```ToLength(? Get(obj, "length"))```
return an integer rather than an abrupt completion. (There's also an expectation that it will have some integer index properties, though that presumably isn't a requirement, i.e. you can have an 'empty' array-like.)

Assuming that's the definition, the `IsSparse` operation doesn't *require* that its `_obj_` be array-like (since it doesn't care where `_len_` came from), but on the other hand, the spec will only call the operation with an array-like object. Maybe the operation should only have the `_obj_`  parameter, and 'recalculate' `_len_`.  That would make it more self-contained and more deserving of "ArrayLike" in the name.

Maybe there should be a `GetLengthOfArrayLike` operation, and maybe that should be used wherever an array-like is required. (E.g., that would help convey the generic-ness of `Array.prototype.*`.) That (and defining "array-like") sounds like a separate PR though.

Reply:
I’d consider it already defined based on the abstract operation you mentioned, just not explicitly.